tests/supply_chain/test_env_reset.py

```diff
@@ -0,0 +1,474 @@
+
+        consumer_node_index = sku3_consumer_unit.data_model_index
+
+        features = ("id", "facility_id", "sku_id", "order_base_cost", "purchased", "received", "order_product_cost",
+                    "latest_consumptions", "in_transit_quantity")
+
+        # ##################################### Before reset #####################################
+        consumer_nodes = env.snapshot_list["consumer"]
+        action = ConsumerAction(sku3_consumer_unit.id, SKU3_ID, supplier_3.id, 1, "train")
+        expect_tick = 100
+
+        # Save the env.metric of each tick into env_metric_1
+        env_metric_1: Dict[int, dict] = defaultdict(dict)
+
+        # Store the information about the snapshot consumer unit of each tick in states_1
+        states_1: Dict[int, list] = defaultdict(list)
+
+        for i in range(expect_tick):
+            env.step([action])
+            env_metric_1[i] = env.metrics
+            states_1[i] = consumer_nodes[i:consumer_node_index:features].flatten().astype(np.int)
+
+        # ############### Test whether reset updates the consumer unit completely ################
+        env.reset()
+        env.step(None)
+
+        # snapshot should reset after env.reset()
+        states = consumer_nodes[1:consumer_node_index:features].flatten().astype(np.int)
+        self.assertEqual([0, 0, 0, 0, 0, 0, 0, 0, 0], list(states))
+
+        expect_tick = 100
+
+        # Save the env.metric of each tick into env_metric_2
+        env_metric_2: Dict[int, dict] = defaultdict(dict)
+
+        # Store the information about the snapshot consumer unit of each tick in states_2
+        states_2: Dict[int, list] = defaultdict(list)
+        for i in range(expect_tick):
+            env.step([action])
+            env_metric_2[i] = env.metrics
+            states_2[i] = consumer_nodes[i:consumer_node_index:features].flatten().astype(np.int)
+
+        expect_tick = 100
+        for i in range(expect_tick):
+            self.assertEqual(list(states_1[i]), list(states_2[i]))
+            self.assertEqual(list(env_metric_1[i].values()), list(env_metric_2[i].values()))
+
+    def test_distribution_unit_reset(self) -> None:
+        """Test initial state of the DistributionUnit of Supplier_SKU3.Test distribution unit reset"""
+        env = build_env("case_02", 100)
+        be = env.business_engine
+        assert isinstance(be, SupplyChainBusinessEngine)
+
+        env.step(None)
+
+        supplier_3 = be.world._get_facility_by_name("Supplier_SKU3")
+        warehouse_1 = be.world._get_facility_by_name("Warehouse_001")
+
+        distribution_unit = supplier_3.distribution
+        distribution_node_index = distribution_unit.data_model_index
+        distribution_nodes = env.snapshot_list["distribution"]
+
+        features = ("id", "facility_id", "pending_order_number", "pending_product_quantity")
+
+        # ##################################### Before reset #####################################
+
+        order_1 = Order(
```

Comment:
why do we place specific order_1, order_2, order_3, and also orders in random ticks?

Reply:
The idea I set up like this is that order_1, order_2, order_3, are their own unit functions. Then the random ticks later is because, I think the current order needs to be passed in the current env.tick, which will affect the expected_finish_tick. So the purpose is to randomly tick several orders in the set rounds to make the test more perfect.